sdk/provisioning/Azure.Provisioning.WebPubSub/src/WebPubSubService.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Provisioning.ResourceManager;
+using Azure.ResourceManager.WebPubSub;
+using Azure.ResourceManager.WebPubSub.Models;
+
+namespace Azure.Provisioning.WebPubSub
+{
+    /// <summary>
+    /// Represents a WebPubSub.
+    /// </summary>
+    public class WebPubSubService : Resource<WebPubSubData>
+    {
+        // https://learn.microsoft.com/azure/templates/microsoft.signalrservice/2023-02-01/webPubSub?pivots=deployment-language-bicep
+        private const string ResourceTypeName = "Microsoft.SignalRService/webPubSub";
+        // https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/webpubsub/Azure.ResourceManager.WebPubSub/src/Generated/RestOperations/WebPubSubRestOperations.cs#L36
+        internal const string DefaultVersion = "2021-10-01";
+
+        private static WebPubSubData Empty(string name) => ArmWebPubSubModelFactory.WebPubSubData();
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="WebPubSubService"/> class.
+        /// </summary>
+        /// <param name="scope">The scope.</param>
+        /// <param name="sku">The SKU.</param>
+        /// <param name="parent">The parent.</param>
+        /// <param name="name">The name.</param>
+        /// <param name="version">The version.</param>
+        /// <param name="location">The location.</param>
+        public WebPubSubService(
+            IConstruct scope,
+            BillingInfoSku? sku = default,
+            ResourceGroup? parent = default,
+            string name = "WebPubSub",
```

Comment:
This will end up being lower case based on the call to GetGloballyUniqueName so may as well just default the lowercase version here.